NitroxClient/Communication/Packets/Processors/InitialPlayerSyncProcessor.cs

```diff
@@ -438,6 +428,62 @@ public void CreateRemotePlayers(object sender, EventArgs eventArgs)
             }
         }
 
+        private void SpawnVehiclesAfterBasePiecesFinish(List<VehicleModel> vehicleModels, bool basePiecesToSpawn)
+        {
+            Log.Info("Received initial sync packet with " + vehicleModels.Count + " vehicles.");
+
+            if (vehicleModels.Count == 0)
+            {
+                return;
+            }
+
+            VehicleCreator vehicleCreator = new VehicleCreator(vehicleModels, vehicles);
+
+            if(basePiecesToSpawn)
+            {
+                ThrottledBuilder.main.QueueDrained += vehicleCreator.CreateVehicles;
+            }
+            else
+            {
+                vehicleCreator.CreateVehicles(null, null);
+            }
+        }
+
+        private class VehicleCreator
+        {
+            private List<VehicleModel> vehicleModels;
+            private Vehicles vehicles;
+
+            public VehicleCreator(List<VehicleModel> vehicleModels, Vehicles vehicles)
+            {
+                this.vehicleModels = vehicleModels;
+                this.vehicles = vehicles;
+            }
+
+            public void CreateVehicles(object sender, EventArgs eventArgs)
+            {
+                ThrottledBuilder.main.QueueDrained -= CreateVehicles;
+
+                List<VehicleModel> cyclopses = vehicleModels
+                    .Where(x => x.TechType == TechType.Cyclops)
+                    .ToList();
+
+                List<VehicleModel> nonCyclopses = vehicleModels
+                    .Where(x => !cyclopses.Any(x2 => x2.TechType == x2.TechType))
+                    .ToList();
+
+                foreach(VehicleModel vehicle in cyclopses)
+                {
+                    vehicles.CreateVehicle(vehicle);
```

Comment:
If both nonCyclopses and Cyclopses need to call `CreateVehicle` then why not just iterate on `vehicleModels` without filtering for `nonCyclopses` and `cyclopses`?

Reply:
This is to spawn the cyclopses first, as a non-cyclops could be docked inside a cyclops.

Reply:
Leave some code comments to explain this for future reference.

Reply:
Do we need to leave frames between spawning cyclops and non-cyclops? From my understanding, cyclops is usually spawned by including a unity scene that performs the build animation (even a brief one when `sub` is used)

Reply:
Is it necessary to create two in-memory lists (`ToList()`) of these two queries, or can they be left as `IEnumerable`s that do the filtering when they are iterated?